src/commands/convert.js

```diff
@@ -16,64 +16,120 @@ const convert = (context, appid, location) => {
-    const legacyDumpUrl = `${
-      constants.BASE_ENDPOINT
-    }/api/developer/v1/apps/${appid}/dump`;
-    const cliDumpUrl = `${
-      constants.BASE_ENDPOINT
-    }/api/developer/v1/apps/${appid}/cli-dump`;
+    if (isVisual) {
+      // has info about the app, such as title
+      const appInfoUrl = `${
+        constants.BASE_ENDPOINT
+      }/api/platform/cli/apps/${appid}`;
+      // has a CLI version of the actual app implementation
+      const versionInfoUrl = `${
+        constants.BASE_ENDPOINT
+      }/api/platform/cli/apps/${appid}/versions/${context.argOpts.version}`;
 
-    utils.startSpinner('Downloading app from Zapier');
+      utils.startSpinner('Downloading app from Zapier');
+      let [appInfo, versionInfo] = [];
+      try {
+        [appInfo, versionInfo] = await Promise.all([
+          utils.callAPI(null, { url: appInfoUrl }, true),
+          utils.callAPI(null, { url: versionInfoUrl }, true)
+        ]);
+      } catch (e) {
+        if (e.status === 404) {
+          throw new Error(
+            `Visual Builder app ${appid} @ ${
+              context.argOpts.version
+            } not found. If you want to convert a Legacy Web Builder app, don't pass a \`--version\` option.`
+          );
+        }
+      }
+      utils.endSpinner();
+      return utils.convertVisualApp(appInfo, versionInfo, tempAppDir);
+    } else {
+      // has info about the app, such as title
+      const legacyDumpUrl = `${
+        constants.BASE_ENDPOINT
+      }/api/developer/v1/apps/${appid}/dump`;
+      // has a CLI version of the actual app implementation
+      const cliDumpUrl = `${
+        constants.BASE_ENDPOINT
+      }/api/developer/v1/apps/${appid}/cli-dump`;
 
-    const [legacyApp, appDefinition] = await Promise.all([
-      utils.callAPI(null, { url: legacyDumpUrl }),
-      utils.callAPI(null, { url: cliDumpUrl })
-    ]);
+      utils.startSpinner('Downloading app from Zapier');
+      let [legacyApp, appDefinition] = [];
+      try {
+        [legacyApp, appDefinition] = await Promise.all([
+          utils.callAPI(null, { url: legacyDumpUrl }, true),
+          utils.callAPI(null, { url: cliDumpUrl }, true)
+        ]);
+      } catch (e) {
+        if (e.status === 404) {
+          throw new Error(
+            `Legacy Web Builder app ${appid} not found. If you want to convert a Visual Builder app, make sure to pass a \`--version\` option.`
+          );
+        }
+      }
 
-    // The JSON dump of the app doesn't have app ID, let's add it here
-    legacyApp.general.app_id = appid;
+      // The JSON dump of the app doesn't have app ID, let's add it here
+      legacyApp.general.app_id = appid;
 
-    utils.endSpinner();
+      utils.endSpinner();
 
-    return await utils.convertApp(legacyApp, appDefinition, tempAppDir);
+      return utils.convertLegacyApp(legacyApp, appDefinition, tempAppDir);
+    }
   };
 
   return utils.initApp(context, location, createApp).then(() => {
```

Comment:
Why not instead of asking for the project location, we use the same string that we put in `name` for the `package.json`? (kebab-cased app title) I think it'd be rare that a dev would want those names to be different.

Reply:
mostly so that devs can run the command from anywhere. `zapier convert 1234 ../../projects`. Since we're writing (and potentially overwriting) files, seems good to be explicit about it.

Reply:
I hear you. I wonder if that's what most do?

It'd be nice to have the option to just run `zapier convert`, get a list of integrations, pick the integration I want, and confirm things like version and/or location. Not for this PR—just thinking out loud.

Reply:
totally. I think this is good to get it out and we can revisit it at some point.